provider/queue/queue.go

```diff
@@ -124,7 +80,26 @@ func (q *Queue) work() {
 
 		for {
 			if c == cid.Undef {
-				k, c = q.nextEntry()
+				head, e := q.getQueueHead()
```

Comment:
Ideally, we'd:

1. Open a query.
2. Iterate over the query to the end.
3. Open a new query when we get the next CID.

(but we can punt on that)

Reply:
I got you. I'll keep that in mind.